### PR TITLE
Add report filter for only the pages a user is assigned to

### PR DESCRIPTION
**Pull request by flashbackzoo**
__Default__

filter-off

__Filter applied__

filter-on

Also fixed up the broken docs link in readme.

### CODE REVIEW

File: code/reports/PagesDueForReviewReport.php [hunk 1]
```diff
@@ -34,6 +34,8 @@ public function parameterFields()
 
         $filtersList->push(new CheckboxField("ShowVirtualPages", "Show Virtual Pages"));
 
+        $filtersList->push(new CheckboxField("OnlyMyPages", "Only Show pages assigned to me"));
```
Comment (scott1702) [inline code/reports/PagesDueForReviewReport.php]: This might need translating :)

File: code/reports/PagesDueForReviewReport.php [hunk 2]
```diff
@@ -153,6 +155,11 @@ public function columns()
             ));
         }
 
+        // Only show pages assigned to the current user?
+        if (!empty($params["OnlyMyPages"])) {
+            $records = $records->filter('ContentReviewUsers.ID', Member::currentUserID());
```
Comment (tractorcow) [inline code/reports/PagesDueForReviewReport.php]: This won't work if I'm in a group that's a reviewer of this page only. It needs to handle both I'm afraid.

You can solve this by first evaluating all group IDs that the current user is in, then using filterAny() on both ContentReviewUsers.ID and ContentReviewGroups.ID

Comment (flashbackzoo) [inline code/reports/PagesDueForReviewReport.php]: Ahh of course. Thanks :smile:

**Comment by flashbackzoo** [comment]
@tractorcow @scott1702 updated :smile:

**Comment by tractorcow** [comment]
I found a bug where "inherit" is used as a content review setting for a page. The filter only works with directly assigned pages.

It should use the same logic as in `ContentReviewEmails::getOverduePagesForOwners`

You can fix this by moving the `"OnlyMyPages"` filter to the end of the list.


```php
// Only show pages assigned to the current user?
if (!empty($params["OnlyMyPages"])) {
	$currentUser = Member::currentUser();
	$records = $records->filterByCallback(function($page) use ($currentUser) {
		$option = $page->getOptions();
		foreach ($option->ContentReviewOwners() as $owner) {
			if($currentUser->ID == $owner->ID) {
				return true;
			}
		}
		return false;
	});
}
```

**Comment by tractorcow** [comment]
It works fine; Merge once the tests pass. :)

**Pull request merged**